cdap-app-fabric/src/main/java/co/cask/cdap/app/preview/PreviewRunner.java

```diff
@@ -0,0 +1,77 @@
+/*
+ * Copyright © 2016 Cask Data, Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License"); you may not
+ * use this file except in compliance with the License. You may obtain a copy of
+ * the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+ * WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
+ * License for the specific language governing permissions and limitations under
+ * the License.
+ */
+
+package co.cask.cdap.app.preview;
+
+import co.cask.cdap.api.metrics.MetricTimeSeries;
+import co.cask.cdap.proto.id.ApplicationId;
+import com.google.gson.JsonElement;
+import org.apache.twill.api.logging.LogEntry;
+
+import java.util.Collection;
+import java.util.List;
+import java.util.Map;
+
+/**
+ * Interface responsible for managing the lifecycle of a single preview application
+ * and retrieving the data generated by that application.
+ */
+public interface PreviewRunner {
+
+  /**
+   * Start the preview of an application.
+   * @param request the {@link PreviewRequest} with which preview to be started
+   * @throws Exception if there were any error during starting preview
+   */
+  void start(PreviewRequest<?> request) throws Exception;
+
+  /**
+   * Get the status of the preview represented by this {@link PreviewRunner}.
+   * @return the status associated with the preview
+   */
+  PreviewStatus getStatus();
+
+  /**
+   * Stop the preview run represented by this {@link PreviewRunner}.
+   * @throws Exception if there were any error during stopping preview
+   */
+  void stop() throws Exception;
+
+  /**
+   * Get list of tracers used in the preview run represented by this {@link PreviewRunner}.
+   * @return {@link List} of tracers used in the preview
+   */
+  List<String> getTracers();
+
+  /**
+   * Get the data associated with the preview run represented by this {@link PreviewRunner}.
+   * @param tracerName the name of the tracer used for preview
+   * @return the {@link Map} of properties associated with the tracer for a given preview
+   */
+  Map<String, List<JsonElement>> getData(String tracerName);
+
+  /**
+   * Get metric associated with the preview run represented by this {@link PreviewRunner}.
+   * @return the {@link Collection} of metrics emitted during the preview run
+   */
+  Collection<MetricTimeSeries> getMetrics();
```

Comment:
How do you decide when to return `Collection` and when to return `List`? The choice seems a bit random to me. Usually you return a List if things are ordered (e.g. logs, metrics). Then for tracer names, it seems more like it should return a `Set` instead of `List` as there shouldn't be any duplicate.